src/server.tsx

```diff
@@ -44,34 +45,45 @@ interface Response {
     css: string;
     js: string;
     meta: EpicTestTracking;
+    debug?: Debug;
 }
 
 const [, fetchConfiguredEpicTestsCached] = cacheAsync(fetchConfiguredEpicTests, 60);
 
-const asResponse = (component: JsComponent, meta: EpicTestTracking): Response => {
+const asResponse = (component: JsComponent, meta: EpicTestTracking, debug?: Debug): Response => {
     const { el, js } = component;
     const { html, css } = extractCritical(renderToStaticMarkup(el));
-    return { html, css, js, meta };
+    return { html, css, js, meta, debug };
 };
 
 const buildEpic = async (
     pageTracking: EpicPageTracking,
     targeting: EpicTargeting,
+    params: Params,
 ): Promise<Response | null> => {
     const configuredTests = await fetchConfiguredEpicTestsCached();
     const hardcodedTests = await getAllHardcodedTests();
     const tests = [...configuredTests.tests, ...hardcodedTests];
-    const result = findTestAndVariant(tests, targeting);
+
+    let result: Result;
+
+    if (params.force) {
+        const test = tests.find(test => test.name === params.force?.testName);
```

Comment:
Could this be `params.force.testName`? I.e. no `?.`? Since we've already checked that `params.force` is truthy (perhaps it's just needed to keep TypeScript happy?)

Reply:
unfortunately not. I think because it is a functional literal, TS isn't clever enough to know it is always present.